bt_editor/mainwindow.cpp

```diff
@@ -96,14 +103,20 @@ MainWindow::MainWindow(GraphicMode initial_mode, QWidget *parent) :
     ui->leftFrame->layout()->addWidget( _replay_widget );
 
 #ifdef ZMQ_FOUND
-    _monitor_widget = new SidepanelMonitor(this);
+    _monitor_widget = new SidepanelMonitor(
+        this, _monitor_publisher_port, _monitor_server_port);
     ui->leftFrame->layout()->addWidget( _monitor_widget );
 
     connect( ui->toolButtonConnect, &QToolButton::clicked,
             _monitor_widget, &SidepanelMonitor::on_Connect );
 
     connect( _monitor_widget, &SidepanelMonitor::connectionUpdate,
             this, &MainWindow::onConnectionUpdate );
+
+    if (monitor_autoconnect) {
+        usleep(1e6);  // Add a slight delay
```

Comment:
Not super thrilled about this delay, so looking for input here.

Reply:
Not sure about alternatives, but the comment can better state why the delay is required. Now the comment is a less brief version of the code.

Reply:
I was thinking maybe no delay, but to add a configurable timeout argument to the "connect" button callback that is more generous during this autoconnect action vs. manually clicking the button.

Reply:
OK, addressed this!